impala/tests/test_data_types.py

```diff
@@ -49,6 +51,47 @@ def test_cursor_description_precision_scale(cur, decimal_table):
     for (exp, obs) in zip(expected, observed):
         assert exp == obs
 
+
+@fixture(scope='module')
+def decimal_table2(cur):
+    table_name = 'tmp_decimal_table2'
+    ddl = """CREATE TABLE {0} (val decimal(18, 9))""".format(table_name)
+    cur.execute(ddl)
+    cur.execute('''insert into {0}
+                   values (cast(123456789.123456789 as decimal(18, 9))),
+                          (cast(-123456789.123456789 as decimal(18, 9))),
+                          (cast(0.000000001 as decimal(18, 9))),
+                          (cast(-0.000000001 as decimal(18, 9))),
+                          (cast(999999999.999999999 as decimal(18, 9))),
+                          (cast(-999999999.999999999 as decimal(18, 9)))'''.format(table_name))
+    try:
+        yield table_name
+    finally:
+        cur.execute("DROP TABLE {0}".format(table_name))
+
+
+def common_test_decimal(cur, decimal_table):
+    """Read back a few decimal values in a wide range."""
+    cur.execute('select val from {0} order by val'.format(decimal_table))
+    results = cur.fetchall()
+    assert results == [(Decimal('-999999999.999999999'),),
```

Comment:
nice to have: it would increase the coverage of the tests if there would be also NULL values in these test tables

Reply:
Agree, added